cql3/restrictions/statement_restrictions.hh

```diff
@@ -132,6 +132,18 @@ public:
      */
     statement_restrictions(schema_ptr schema, bool allow_filtering);
 
+    friend statement_restrictions analyze_statement_restrictions(
```

Comment:
I think the more traditional approach to such a "factory function" is to make it a static class function, instead of a global friend function, and have it called as, for example, `statement_restictions::analyze(....)` instead of your `analyze_statement_restrictions`.
But I don't insist, I guess it's a matter of style.